### PR TITLE
fix(components/filter-bar): add top border styling for filter bar in data manager

### PR DESCRIPTION
**Pull request by Blackbaud-CoreyArcher**
[AB#3626265](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/3626265)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/4094/storybook/)

Component Storybooks:

- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/4094/storybooks/data-manager-storybook/)
- [filter-bar-storybook](https://blackbaud.github.io/skyux-pr-preview/4094/storybooks/filter-bar-storybook/)

Apps:

- [code-examples-playground](https://blackbaud.github.io/skyux-pr-preview/4094/code-examples-playground/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/4094/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/4094/playground/)

File: libs/components/filter-bar/src/lib/modules/filter-bar/filter-bar.component.scss [hunk 1]
```diff
@@ -11,3 +11,12 @@
...
 }
+
+// When filter bar is a child of data manager toolbar, add a top border as if it were a toolbar section.
+:host-context(.sky-data-manager-toolbar) {
+  border-top: var(
+    --sky-override-toolbar-border-top-bottom,
+    var(--sky-border-width-divider) var(--sky-border-style-divider)
+      var(--sky-color-border-divider)
+  );
+}
```
Comment (Blackbaud-SandhyaRajasabeson) [inline libs/components/filter-bar/src/lib/modules/filter-bar/filter-bar.component.scss:22]: So this visual look is the same as the `border-top` from toolbar for a toolbar section, can we get a visual review from design that that is what we want here? Or is there a comment or discussion somewhere on how we landed on this styling?

For that to happen we would also need a visual test for this and it looks like none of the existing visual tests for toolbar were change from this PR, can we add a visual test?

Comment (Blackbaud-CoreyArcher) [reply libs/components/filter-bar/src/lib/modules/filter-bar/filter-bar.component.scss:22]: The bug says the toolbar should have a section border on top.

Added screenshot of data manager with extra toolbars.

File: apps/e2e/data-manager-storybook-e2e/src/e2e/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.cy.ts [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  E2eVariations.forEachTheme((theme) => {
+    describe(`in ${theme} theme`, () => {
+      beforeEach(() =>
+        cy.visit(
+          `/iframe.html?globals=theme:${theme}&id=data-manager-with-list-toolbarscomponent--data-manager-with-list-toolbars`,
+        ),
+      );
+
+      it('should render the component', () => {
+        cy.get('app-data-manager-with-list-toolbars')
```
Comment (Blackbaud-SandhyaRajasabeson) [inline apps/e2e/data-manager-storybook-e2e/src/e2e/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.cy.ts:14]: can you add `cy.skyReady()` to make sure the icons are loaded before the screenshots

File: apps/e2e/data-manager-storybook/src/app/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+@Component({
+  standalone: false,
+  selector: 'app-data-manager-with-list-toolbars',
+  templateUrl: './data-manager-with-list-toolbars.component.html',
+  styleUrls: ['./data-manager-with-list-toolbars.component.scss'],
+  providers: [SkyDataManagerService],
+})
+export class DataManagerWithListToolbarsComponent implements OnInit {
+  @Input()
+  public activeView = 'view-1';
```
Comment (Blackbaud-SandhyaRajasabeson) [inline apps/e2e/data-manager-storybook/src/app/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.ts]: Can you convert this to a signal instead

File: libs/components/filter-bar/src/lib/modules/filter-bar/filter-bar.component.scss (hunk 1 above)
Comment (Blackbaud-SandhyaRajasabeson) [reply libs/components/filter-bar/src/lib/modules/filter-bar/filter-bar.component.scss:22]: ah i missed what that meant in the "Expected behavior" I've left the new screenshots for design to take a look and approve if necessary jic

File: apps/e2e/data-manager-storybook/src/app/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.ts (hunk 3 above)
Comment (Blackbaud-CoreyArcher) [reply apps/e2e/data-manager-storybook/src/app/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.ts]: this input isn't used. removing

File: apps/e2e/data-manager-storybook-e2e/src/e2e/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.cy.ts (hunk 2 above)
Comment (Blackbaud-CoreyArcher) [reply apps/e2e/data-manager-storybook-e2e/src/e2e/data-manager-with-list-toolbars/data-manager-with-list-toolbars.component.cy.ts:14]: added

**Review by Blackbaud-SandhyaRajasabeson — APPROVED** [review]
